From e1b567f07b87efd8fd8f527242bfbf5b30702d9c Mon Sep 17 00:00:00 2001
From: Vu Van Tu <tuvv@mageplaza.com>
Date: Mon, 31 Aug 2020 15:07:10 +0700
Subject: [PATCH] fix not subscriber on popup

---
 Plugin/Controller/Subscriber/NewAction.php | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/vendor/mageplaza/module-better-popup/Plugin/Controller/Subscriber/NewAction.php b/vendor/mageplaza/module-better-popup/Plugin/Controller/Subscriber/NewAction.php
index d5ea128..3ee32ed 100644
--- a/vendor/mageplaza/module-better-popup/Plugin/Controller/Subscriber/NewAction.php
+++ b/vendor/mageplaza/module-better-popup/Plugin/Controller/Subscriber/NewAction.php
@@ -29,7 +29,9 @@
 use Magento\Framework\Controller\Result\Json;
 use Magento\Framework\Controller\Result\JsonFactory;
 use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Validator\EmailAddress as EmailValidator;
 use Magento\Newsletter\Model\SubscriberFactory;
+use Magento\Newsletter\Model\SubscriptionManagerInterface;
 use Magento\Store\Model\StoreManagerInterface;
 use Mageplaza\BetterPopup\Helper\Data;

@@ -58,6 +60,8 @@ class NewAction extends \Magento\Newsletter\Controller\Subscriber\NewAction
      * @param StoreManagerInterface $storeManager
      * @param CustomerUrl $customerUrl
      * @param CustomerAccountManagement $customerAccountManagement
+     * @param SubscriptionManagerInterface $subscriptionManager
+     * @param EmailValidator|null $emailValidator
      * @param JsonFactory $resultJsonFactory
      * @param Data $helperData
      */
@@ -68,20 +72,16 @@ public function __construct(
         StoreManagerInterface $storeManager,
         CustomerUrl $customerUrl,
         CustomerAccountManagement $customerAccountManagement,
+        SubscriptionManagerInterface $subscriptionManager,
+        EmailValidator $emailValidator = null,
         JsonFactory $resultJsonFactory,
         Data $helperData
     ) {
         $this->resultJsonFactory = $resultJsonFactory;
         $this->_helperData       = $helperData;

-        parent::__construct(
-            $context,
-            $subscriberFactory,
-            $customerSession,
-            $storeManager,
-            $customerUrl,
-            $customerAccountManagement
-        );
+        parent::__construct($context, $subscriberFactory, $customerSession, $storeManager, $customerUrl,
+            $customerAccountManagement, $subscriptionManager, $emailValidator);
     }

     /**
